Increment Copyright Year 2020

JoshuaFern opened the pull request:
**Motivation for change:**
In the roadmap to first release, there's a [card](https://github.com/dreamer/dosbox-staging/projects/2#card-33478874) for "Update copyright dates throughout the project.", is this what you had in mind? We're not lawyers, but do you have any clue if we're allowed to increment copyright dates in this fashion?

**Details**
* First I ran: `rpl -Rvw 2002-2019 2002-2020 ./*`
* After reviewing the changes I noticed `src/winres.rc` was now missing the "�" symbols, I'm guessing this is a copyright symbol but my browser does not render it correctly. I manually re-added them to the file. Also found and corrected similar problem in `dosbox-installer.nsi`. Using `rpl` has proven unreliable, removing symbols that shouldn't be stripped, or maybe I'm just not using it correctly.

Excited for the first release of dosbox-staging, hope this helps. Let me know if you had something else in mind.

JoshuaFern commented:
Great points, copyright is a complicated beast.

dreamer commented:
I was dragging my feet about updating copyright dates because I was waiting for upstream to do it. Usually, DOSBox maintainers updated the copyright date in January, but this year the change still hasn't landed…

As for updating dates in a single file versus all files at once - updating file-by-file is technically more correct, but it results in people forgetting about it. For projects I participated in it usually went like this: people were forgetting to update copyright year, this leads to another thing to check during code review (which is irritating for both author and reviewer), then I try to automate it via script or git hook, then people forget to use it, etc, etc… At the moment I think updating the dates once a year throughout the project is just a more practical solution.

Did a quick check with:

    $ git grep -i copyright | grep 2019 | grep -v 2020

and it seems like there are some additional files to update; most of them are scripts or source files we introduced in 2019. 

Don't update copyright years in `src/libs/` directory though - that should be done on case-by-case basis.

JoshuaFern commented:
>I was dragging my feet about updating copyright dates because I was waiting for upstream to do it. Usually, DOSBox maintainers updated the copyright date in January, but this year the change still hasn't landed…

Should we just ignore the default DOSBox copyright message until it's changed upstream?

>and it seems like there are some additional files to update; most of them are scripts or source files we introduced in 2019.

Would you like me to add 2019-2020 for those new ones?

> Don't update copyright years in `src/libs/` directory though - that should be done on case-by-case basis.

What about `src/libs/zmbv`? This is the only lib I updated the years on so far, but I believe it's one of those special cases you mentioned.

JoshuaFern commented:
I noticed the following: `src/libs/decoders/xxh3.h:   Copyright (C) 2019-present, Yann Collet.`
Are you allowed to do this? If so it would solve a lot of headache manually updating the years every year, but I doubt it's actually correct.

dreamer commented:
> > I was dragging my feet about updating copyright dates because I was waiting for upstream to do it. Usually, DOSBox maintainers updated the copyright date in January, but this year the change still hasn't landed…
> 
> Should we just ignore the default DOSBox copyright message until it's changed upstream?

Nah. Git will handle merging upstream change, no need to worry about it. And we don't want to wait for upstream to fix it before first release of staging.



> >  and it seems like there are some additional files to update; most of them are scripts or source files we introduced in 2019.
>
> Would you like me to add 2019-2020 for those new ones?

Yes, please :)

> What about src/libs/zmbv? 

Technically it's a dependency, but practically it was developed by DOSBox team. Let's update dates in this single dir as well.

> I noticed the following: `src/libs/decoders/xxh3.h: Copyright (C) 2019-present, Yann Collet.`
> Are you allowed to do this? If so it would solve a lot of headache manually updating the years every year, but I doubt it's actually correct.

No, this is not correct. Hopefully, xxHash author will adjust it someday. Leave this line as it is, please.

dreamer commented:
@JoshuaFern thanks for changes, @krcroft thanks for review :)

After CI will pass, I'll squash the changes into a single commit.

JoshuaFern commented:
Yup, this is ready to merge. The rest of the copyrights I'm finding are not to be updated :)

dreamer commented:
Rebased, squashed and merged-in as commit: f2029d71d8d97e54358dbb0610643a879468f22a

@JoshuaFern thanks again :) One step closer to release!

krcroft commented:
@JoshuaFern, did you put together a Python script to help with this? (just guessing based on the reference to another GitHub issue); if so, we could drop it in, say `./contrib/version_update/...`.  It will certainly be handy every year!

JoshuaFern commented:
> @JoshuaFern, did you put together a Python script to help with this? (just guessing based on the reference to another GitHub issue); if so, we could drop it in, say `./contrib/version_update/...`. It will certainly be handy every year!

Originally I used https://github.com/kcoyner/rpl, but I found it was stripping out unicode characters. 
I found this actively-developed fork https://github.com/rrthomas/rpl but I haven't tested it yet.
Most of this work could have been done with a fairly simple sed script instead.

krcroft commented:
Interesting; this is a handy reference for the next go-round in 2021 :-)
